lib/stack.rb

```diff
@@ -1,3 +1,63 @@
+require_relative './node'
```

Comment:
If you want to use node why just not create a module and include it in the class. For example:

```
include DataStructure::Node
```
And see if you just can remove the dependency of the stack class over Node class.